### ISSUE TITLE
cli.Command false positive tests

### ISSUE THREAD

**Issue by johann-koebbe** [vlang/v#19135]
### Describe the bug

The tests for cli.Command contain a lot of false positives. For example, the test
```
fn test_if_subcommands_parse_args() {
     mut cmd := cli.Command{
         name: 'command'
     }
     subcmd := cli.Command{
         name: 'subcommand'
         execute: if_subcommands_parse_args_func
     }
     cmd.add_command(subcmd)
     cmd.parse(['command', 'subcommand', 'arg0', 'arg1'])
 }
```
If I change the last line to 
```
cmd.parse(['command', 'notsubcommand', 'arg0', 'arg1'])
```
and run
```
v test vlib/cli/command_test.v
```
I get no failures. I experimented with a lot of other tests, prepending `not` to the command being parsed, and the tests continued to pass.

### Expected Behavior

I expect the test to fail.

### Current Behavior

No tests fail.

### Reproduction Steps

As noted above.

### Possible Solution

Not sure what the solution is without a way to assert a function was called.

### Additional Information/Context

My suspicion is that the subcommand `execute` callback is not getting called (obviously) and the `cmd.parse` statement exits without error, leaving the test nothing to do but pass. @

### V version

be1270540b6037825bccd3ce6adf0179bc92932d

### Environment details (OS name and version, etc.)

macOS Ventura 13.4.1.

**Comment by sibkod** [comment]
If I understand correctly, then we are not parsing the argument `name: 'subcommand", so the method `if_subcommands_parse_args_func` is not called, so the test ends successfully.
In your case, you should change the check to something like
assert has_command(cmd, 'notsubcommand')

**Comment by johann-koebbe** [comment]
@sibkod Yeah, I see the error in my thinking now. Thanks for pointing that out. :)

I do think, however, there still exists the possibility of a false positive. I might be reaching here, but suppose the execute callback doesn't get called for some reason. The test will end successfully.
